onnx/defs/shape_inference.h

```diff
@@ -102,6 +103,7 @@ struct InferenceContext {
   virtual const SparseTensorProto* getInputSparseData(size_t index) const = 0;
   // Gets the shape inputs computed by partial data propagation.
   virtual const TensorShapeProto* getSymbolicInput(size_t index) const = 0;
+  virtual const ShapeInferenceOptions& getShapeInferenceOptions() const = 0;
```

Comment:
I wonder if we should provide default-handling in this file too (for backward-compatibility)? Eg., we could return a pointer (with the default-implementation returning nullptr). Same for `doInferencing` above.

Reply:
default option used. also removed from DoInferencing the shapeinferenceOption